src/NzbDrone.Core/HealthCheck/Checks/ApiKeyValidationCheck.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Text.RegularExpressions;
+using NLog;
+using NzbDrone.Core.Configuration;
+using NzbDrone.Core.Configuration.Events;
+using NzbDrone.Core.Lifecycle;
+
+namespace NzbDrone.Core.HealthCheck.Checks
+{
+    [CheckOn(typeof(ApplicationStartedEvent))]
+    [CheckOn(typeof(ConfigSavedEvent))]
+    public class ApiKeyValidationCheck : HealthCheckBase
+    {
+        private static readonly Regex ValidateApiKeyRegex = new ("^[a-zA-Z0-9]{20,}$", RegexOptions.Compiled);
+
+        private readonly IConfigFileProvider _configFileProvider;
+        private readonly Logger _logger;
+
+        public ApiKeyValidationCheck(IConfigFileProvider configFileProvider, Logger logger)
+        {
+            _configFileProvider = configFileProvider;
+            _logger = logger;
+        }
+
+        public override HealthCheck Check()
+        {
+            if (!ValidateApiKeyRegex.IsMatch(_configFileProvider.ApiKey))
+            {
+                _logger.Warn("Please consider using a valid or more secure API Key in your configuration");
```

Comment:
If we just focus on length we could change this to something like:

`Please update your API key to be at least 20 characters long. You can do this via settings or the config file`